src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php

```diff
@@ -112,6 +113,17 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $this->validateConfiguration($extension, $configuration);
 
         $format = $input->getOption('format');
+
+        if ('yaml' === $format && !class_exists(Yaml::class)) {
+            $errorIo->error('Setting the "format" option to "yaml" requires the "Yaml" component. Either install "symfony/yaml" or use "xml".');
+
+            return 1;
+        }
+
+        if (null === $format) {
```

Comment:
Does that actually work? The `format` option defaults to `yaml`, so we shouldn't get `null` here. Or am I missing something?

What I was thinking about: We could change the default, depending on the availability of the YAML component.

```php
new InputOption('format', null, InputOption::VALUE_REQUIRED, 'The output format (yaml or xml)', class_exists(Yaml::class) ? 'yaml' : 'xml'),
```

Reply:
> Does that actually work? The format option defaults to yaml, so we shouldn't get null here. Or am I missing something?

You are right. I was missing something. The plan was to remove the `format` options default value like @ chalasr suggested, but somehow I forgot to push this commit.

Reply:
> What I was thinking about: We could change the default, depending on the availability of the YAML component.

Yes, that's a much better solution. Will change it. Thanks @derrabus